src/clusterfuzz/_internal/cron/external_testcase_reader.py

```diff
@@ -0,0 +1,149 @@
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Automated ingestion of testcases via IssueTracker."""
+
+import re
+
+import requests
+
+from appengine.libs import form
+from appengine.libs import gcs
+from appengine.libs import helpers
+from clusterfuzz._internal.issue_management.google_issue_tracker import \
+    issue_tracker
+
+ACCEPTED_FILETYPES = [
+    'text/javascript', 'application/pdf', 'text/html', 'application/zip'
+]
+
+
+def close_invalid_issue(upload_request, attachment_info, description):
+  """Closes any invalid upload requests with a helpful message."""
+  comment_messsage = (
+      'Hello, this issue is automatically closed. Please file a new bug after'
+      'fixing the following issues:\n\n')
+  invalid = False
+
+  # TODO(pgrace) remove after testing.
+  if upload_request.id == '373893311':
+    return False
+
+  # TODO(pgrace) add secondary check for authorized reporters.
+
+  # Issue must have exactly one attachment.
+  if len(attachment_info) != 1:
+    comment_messsage += 'Please provide exactly one attachment.\n'
+    invalid = True
+  else:
+    # Issue must use one of the supported testcase file types.
+    if attachment_info[0]['contentType'] not in ACCEPTED_FILETYPES:
+      comment_messsage += (
+          'Please provide an attachment of type: html, js, pdf, or zip.\n')
+      invalid = True
+    if not attachment_info[0]['attachmentDataRef'] or \
```

Comment:
Can you use paren instead of backslash for continuing lines?

Reply:
ugh damn sorry this was loaded from stashed changes and i have the memory of a goldfish